variables.tf

```diff
@@ -0,0 +1,50 @@
+//  https://azure.microsoft.com/en-ca/global-infrastructure/geographies/
+variable "location" {
+  description = "Name of the location where the resources will be provisioned."
+  type = string
+}
+
+variable "source_management_group" {
+  description = "The source management group from which the Subscriptions will be moved from. The name used here is not the display name, it is the ID shown next to the display name in the Azure Portal Management Group view."
+  type = object({
+    name = string
+    id = string
+  })
+
+  # if the input for the name is incorrect, the default error is explanatory enough
+}
+
+variable "target_management_group" {
+  description = "The target management group to which the subscriptions will be moved. The name used here is not the display name, it is the ID shown next to the display name in the Azure Portal Management Group view."
```

Comment:
id name = id, what is id then?

Reply:
the id is the path from providers/Microsoft.ManagementGroups/<ManagementGroupName(ID)>